### ISSUE TITLE
After restart, chronos doesn't detect pre-restart task until 20 minutes in

### ISSUE THREAD

**Issue by ramkandasamy** [mesos/chronos#659]
Hi guys,

    So we have a Chronos task (it should run every 15 min) that is launched but then is hanging for a long time. When we restart chronos, the task is re-launched, and then about 20 min later, is launched again. Finally, before the 3rd run after the restart can start, Chronos detects the hanging task and does not launch any more tasks.

     Just to be clear, the issue isn't that the Chronos task is hanging. The issue we would like to see fixed is that Chronos should detect previously running tasks on startup, rather than doing a couple runs and then stopping. The reason why it isn't doing it currently is because in MesosJobFramework, we have:

  private var lastReconciliation = DateTime.now.plusSeconds(config.reconciliationInterval())

and we have this reconcile code:

  def reconcile(schedulerDriver: SchedulerDriver): Unit = {
    if (DateTime.now().isAfter(lastReconciliation.plusSeconds(config.reconciliationInterval()))) {
      lastReconciliation = DateTime.now()

      val taskStatuses = runningTasks.keys.flatMap(id => runningTasks.get(id))

      log.info("Performing task reconciliation with the Mesos master")
      schedulerDriver.reconcileTasks(taskStatuses.flatMap(task => task.taskStatus).asJavaCollection)
    }
  }

So basically, the config.reconciliationInterval value is 10 min by default. So we have 10 min + 10 min, so a little over 20 min in, the chronos framework reconciles the tasks with the Mesos framework and then it figures out that there is the hanging task still running. We think that Chronos should do the schedulerDriver.reconcileTasks method on startup, so that way we don't get this unpredictable behavior.

As a workaround, we could make config.reconciliationInterval very small, but it still wouldn't stop a run from going ahead on startup. Also, if we made it too small, we would be making many calls to reconcileTasks, which seems unnecessary.
